### ISSUE TITLE
Select menu doesn't support search option

### ISSUE THREAD

**Issue by MirelaPaun** [Tradeshift/tradeshift-ui#584]
### Bug report
Select menu doesn't support search option/input

#### Tradeshift UI version affected
v10.0.17

#### Expected Behavior
When using `select` menu, a search option is necessary if the list of items is too long.

#### Actual Behavior
When using `select` menu, you can't add the option to search inside the options list.

#### Steps to reproduce
Use `select` menu from http://ui.tradeshift.com/v10/#components/forms/  and try to have a search input before all options. You can't have this feature.

#### Screenshots (optional)
Screenshots are from `DocumentManager` app for `currency` list, which is a long list and the search option is really necessary.

<img width="288" alt="currency_input" src="https://user-images.githubusercontent.com/28289435/37823040-f3954eb6-2e90-11e8-921f-7aca525b066c.png">

<img width="325" alt="currency_list" src="https://user-images.githubusercontent.com/28289435/37823042-f3b19850-2e90-11e8-9fa2-4cd2d8a09406.png">

**Comment by wiredearp** [comment]
cc @mihneaz :beer:

**Comment by mihneaz** [comment]
Yes, I wrote some observations (I initially discussed with @wiredearp) and a few nice to have’s we should take into consideration:
[https://docs.google.com/document/d/1QxxQcvLSjPon7KnerIIPEBsGpWJeBKuRw-7_fa-QdD0/edit?usp=sharing](https://docs.google.com/document/d/1QxxQcvLSjPon7KnerIIPEBsGpWJeBKuRw-7_fa-QdD0/edit?usp=sharing)

@jerf can you review and decide the best course of action?

**Comment by jamessimo** [comment]
We are looking to making something like this in future as well. 
https://tradeshift.invisionapp.com/share/CPFR7DDVT7D#/276586304_Unit_Search
